src/gui/symbology/qgssvgselectorwidget.h

```diff
@@ -410,6 +412,38 @@ class GUI_EXPORT QgsSvgSelectorWidget : public QWidget, private Ui::WidgetSvgSel
      */
     bool allowParamerters() const {return mAllowParameters;}
 
+    /**
+     * Defines if the SVG browser should be visible
+     * \since QGIS 3.20
+     */
+    void setBrowserVisible( bool visible );
+
+    /**
+     * Returns if the SVG browser should be visible
+     * \since QGIS 3.20
+     */
+    bool browserVisible() const {return mBrowserVisible;}
+
+    /**
+     * Defines if the widget should allow any raster or SVG image
+     * or if the file chooser will be limited to open SVG files
+     * \since QGIS 3.20
+     */
+    void setAllowAnyImage( bool allowAnyImage );
+
+    /**
+     * Returns if the widget should allow any raster or SVG image
+     * or if the file chooser will be limited to open SVG files
+     * \since QGIS 3.20
+     */
+    bool allowAnyImage() const {return mAllowAnyImage;}
```

Comment:
Wondering if we could improve a bit here, like setAllowedImageType( ImageTypes type ), with an enum that'd be SVG, Raster, All.

Reply:
I am not sure this makes sense here. The filter is set to any image format including SVG.
I didn't want to put to much logic in here (but auto-hiding SVG specific UX) since the format (SVG/raster/unknown) is auto-determined in picture layout.